src/streamlink/plugins/dailymotion.py

```diff
@@ -92,138 +46,24 @@
-        playlist = http.json(res, schema=_vod_playlist_schema)
-
-        parsed = urlparse(template)
-        url_template = "{0}://{1}{2}".format(
-            parsed.scheme, parsed.netloc, playlist["template"]
-        )
-        segment_max = reduce(lambda i, j: i + j[0], playlist["fragments"], 0)
-
-        substreams = [HTTPStream(self.session,
-                                 url_template.replace("$fragment$", str(i)))
-                      for i in range(1, segment_max + 1)]
-
-        return FLVPlaylist(self.session,
-                           duration=playlist["duration"],
-                           flatten_timestamps=True,
-                           skip_header=True,
-                           streams=substreams)
-
-    def _get_vod_streams(self, params):
-        manifest_url = params.get("autoURL")
-        if not manifest_url:
-            return
-
-        res = http.get(manifest_url)
-        if res.headers.get("Content-Type") == "application/f4m+xml":
-            streams = HDSStream.parse_manifest(self.session, res.url)
-
-            # TODO: Replace with "yield from" when dropping Python 2.
-            for __ in streams.items():
-                yield __
-        elif res.headers.get("Content-Type") == "application/vnd.apple.mpegurl":
-            streams = HLSStream.parse_variant_playlist(self.session, res.url)
-
-            # TODO: Replace with "yield from" when dropping Python 2.
-            for __ in streams.items():
-                yield __
-        else:
-            manifest = http.json(res, schema=_vod_manifest_schema)
-            for params in manifest["alternates"]:
-                name = "{0}p".format(params["height"])
-                stream = self._create_flv_playlist(params["template"])
-                yield name, stream
-
-                failovers = params.get("failover", [])
-                for failover in failovers:
-                    stream = self._create_flv_playlist(failover)
-                    yield name, stream
-
-    def _chrome_cast_stream_fallback(self):
-        self.logger.debug("Trying to find Chromecast URL as a fallback")
-        # get the page if not already available
-        page = http.get(self.url, cookies=COOKIES)
-        m = chromecast_re.search(page.text)
-        if m:
-            url = json.loads(m.group("url"))
-            return HLSStream.parse_variant_playlist(self.session, url)
+        for quality, streams in media['qualities'].items():
+            for stream in streams:
+                if stream['type'] == 'application/x-mpegURL':
+                    if quality != 'auto':
+                        # Avoid duplicate HLS streams with bitrate selector in the URL query
+                        continue
+                    for s in HLSStream.parse_variant_playlist(self.session, stream['url']).items():
+                        yield s
+                elif stream['type'] == 'video/mp4':
+                    bitrate = quality + 'p'
```

Comment:
_this is not so important but_
**bitrate** is currently the **resolution**,
maybe change it to `resolution` , `name` or something

---

`$ streamlink  https://www.dailymotion.com/video/x5qnozs 1080 -l debug`

currently the names of bitrate are
```
144p
240p
380p
480p
720@60p
1080@60p
```
but @ is invalid so it will be 720 and 1080

```
[cli][info] Available streams: 480p_http, 720, 1080, 112p_alt (worst),
112p, 144p, 184p_alt, 184p, 240p, 288p_alt, 288p, 380p, 480p_alt,
480p, 720p_alt, 720p, 1080p_alt, 1080p (best)
```

---

maybe use `re.sub` and replace @ with p
or
use only the resolution and remove the FPS

Reply:
Good catch ;).
I suppose @[0-9]+ corresponds to a FPS.